### PR TITLE
add mogran logger, nodemon and rename interceptors to middlewares

### LINKED ISSUE #232: `Error: Please connect milvus first` should be removed on the initialization of the server
image

### PR DESCRIPTION
**Pull request by nameczz**
Signed-off-by: nameczz <zizhao.chen@zilliz.com>
fix #232 
image

### CODE REVIEW

File: express/package.json [hunk 1]
```diff
@@ -4,12 +4,16 @@
   "main": "dist/app.js",
   "license": "MIT",
   "dependencies": {
+    "@types/chalk": "^2.2.0",
```
Comment (shanghaikid) [inline express/package.json:7]: devDependencies

File: express/src/middlewares/index.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  res: Response,
+  next: NextFunction
+) => {
+  const oldSend = res.json;
+  res.json = (data) => {
+    // console.log(data); // do something with the data
+    const statusCode = data?.statusCode;
+    const message = data?.message;
+    const error = data?.error;
+    res.json = oldSend; // set function back to avoid the 'double-send'
```
Comment (shanghaikid) [inline express/src/middlewares/index.ts:16]: overwrite self?

File: express/src/middlewares/index.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    if (statusCode || message || error) {
+      return res.json({ statusCode, message, error });
+    }
+    return res.json({ data, statusCode: 200 }); // just call as normal with data
+  };
+  next();
+};
+
+const getDurationInMilliseconds = (start: any) => {
```
Comment (shanghaikid) [inline express/src/middlewares/index.ts:25]: If we don't need it , just remove it.

**Review by shanghaikid — CHANGES_REQUESTED** [review]

**Review by czhen-zilliz — APPROVED** [review]

File: express/src/middlewares/index.ts (hunk 2 above)
Comment (nameczz) [reply express/src/middlewares/index.ts:16]: @czhen-zilliz express looks like need overwrite res.json to do response format

Comment (czhen-zilliz) [reply express/src/middlewares/index.ts:16]: Overwrite res.json/res.send to be consistent with nest.js response:
1. Format data in res like: `{ data: <raw data>, statusCode: 200 }`.
2. Format error data like: `{ error: <error details>, message: <error messages>, statusCode: 500 }`.

**Review by shanghaikid — APPROVED** [review]

**Pull request merged**